File: src/main/java/me/danetnaverno/editoni/blockstate/MinecraftSignBlockState.java
package me.danetnaverno.editoni.blockstate;

import me.danetnaverno.editoni.world.BlockState;
import net.querz.nbt.tag.CompoundTag;

public class MinecraftSignBlockState extends BlockState
{
    public final String facing;
    public final boolean waterlogged;

    public MinecraftSignBlockState(CompoundTag properties)
    {
        super(properties);
        facing = properties.getString("facing");
        waterlogged = properties.getBoolean("waterlogged");
    }
}

File: src/main/kotlin/me/danetnaverno/editoni/blockstate/MinecraftSignBlockState.kt
package me.danetnaverno.editoni.blockstate

import net.querz.nbt.tag.CompoundTag

class MinecraftSignBlockState(properties: CompoundTag) : BlockState(properties)
{
    val facing = properties.getString("facing")
    val waterlogged = properties.getBoolean("waterlogged")

}